src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Truck.class) {
+            return (List<T>) new TruckProducer().get();
+        }
+        if (type == Bulldozer.class) {
+            return (List<T>) new BulldozerProducer().get();
+        }
+        if (type == Excavator.class) {
+            return (List<T>) new ExcavatorProducer().get();
+        }
```

Comment:
The current implementation uses type checks to determine which producer to use. This approach works but can become cumbersome if more machine types are added. Consider using a map or another design pattern to handle this more dynamically.